pycbuf/CBufParserPy.cpp

```diff
@@ -587,9 +580,13 @@ bool CBufParserPy::FillPyObjectInternal(uint64_t hash, ast_struct* st, PyObject*
     pypre->variant = pre->variant();
     pypre->type_name = st->name;
     pypre->source_name = source_cbuf_file_;
-    VLOG_ASSERT(pre->hash == hash,
-                "Hash mismatch decoding type %s, expected " U64_FORMAT_HEX ", got " U64_FORMAT_HEX, st->name,
-                hash, pre->hash);
+    if (pre->hash != hash) {
+      PyErr_Format(PyExc_ValueError,
```

Comment:
This is good, does this set the error flag on Python?

Reply:
Yes it does. I'm now using this everywhere